File: programmers/etc/level1/가장_가까운_같은_글자.py
def solution(s):
    answer = []
    d = {}
    for idx, w in enumerate(s):
        if w not in d:
            answer.append(-1)
            d[w] = idx
        else:
            answer.append(idx-d[w])
            d[w] = idx
    return answer


s = "banana"
print(solution(s)) # [-1, -1, -1, 2, 2, 2]

s = "foobar"
print(solution(s)) # [-1, -1, 1, -1, -1, -1]

File: Java/programmers/etc/level1/가장가까운같은글자.java
package CodeTest.Java.programmers.etc.level1;

import java.util.*;

public class 가장가까운같은글자 {
    public List<Integer> solution(String s) {
        List<Integer> answer = new ArrayList<>();
        HashMap<String, Integer> d = new HashMap<>();
        
        String[] st = s.split("");
        for (int idx=0; idx<st.length; idx++) {
            String w = st[idx];
            if (d.containsKey(w)) {
                answer.add(idx-d.get(w));
                d.put(w, idx);
            } else {
                answer.add(-1);
                d.put(w, idx);
            }
        }
        
        return answer;
    }
}
